Sources/TuistSigning/SigningFilesLocator.swift

```diff
@@ -0,0 +1,81 @@
+import TSCBasic
+import TuistCore
+import TuistSupport
+
+enum SigningFilesLocatorError: FatalError {
+    case signingDirectoryNotFound(AbsolutePath)
+
+    var type: ErrorType {
+        switch self {
+        case .signingDirectoryNotFound:
+            return .abort
+        }
+    }
+
+    var description: String {
+        switch self {
+        case let .signingDirectoryNotFound(fromPath):
+            return "Could not find signing directory from \(fromPath.pathString)"
+        }
+    }
+}
+
+protocol SigningFilesLocating {
+    func locateSigningDirectory(at path: AbsolutePath) throws -> AbsolutePath?
+    func locateProvisioningProfiles(at path: AbsolutePath) throws -> [AbsolutePath]
+    func locateUnencryptedCertificates(at path: AbsolutePath) throws -> [AbsolutePath]
+    func locateEncryptedCertificates(at path: AbsolutePath) throws -> [AbsolutePath]
+    func locateUnencryptedPrivateKeys(at path: AbsolutePath) throws -> [AbsolutePath]
+    func locateEncryptedPrivateKeys(at path: AbsolutePath) throws -> [AbsolutePath]
```

Comment:
I think it'd be more accurate to say `from path:` than `at path:` because from implies that we are doing a lookup from there.